### DISCUSSION TITLE
[@types/three] Can't import loaders (STLloader) from version 150 and onwards

### DISCUSSION THREAD

**Discussion by LeunensMichiel** [DefinitelyTyped/DefinitelyTyped#67996] [Issues with a @types package]
import { STLLoader } from "three/examples/jsm/loaders/STLLoader";

I have to downgrade to version 149 to make this work again. Three JS still has STLLoader under the same file directory so I don't think anything has changed in the library.

[a comment by an automated account is not shown]

**Comment by Methuselah96** [comment]
- What error do you get?
- What `"moduleResolution"` are you using?
- Have you tried importing from `three/addons/loaders/STLLoader.js` (as shown in the addons section of [the installation docs](https://threejs.org/docs/index.html#manual/en/introduction/Installation))?

**Reply by Methuselah96** [reply]
We added the `"exports"` to `package.json` in @types/three@0.154.0, so that doesn't quite line up as an explanation. If you could provide a TypeScript playground or a sample repo showing the issue, that would be most helpful. I can't think of why this would be happening when upgrading to r150.

**Reply by LeunensMichiel** [reply]
> Cannot find module 'three/examples/jsm/loaders/STLLoader' or its corresponding type declarations

I'm using `"moduleResolution": "bundler",`. Indeed, when I'm using a codesandbox minimal example, I can't seem to reproduce this. However, it doesn't make sense that it works on version <150 with `"moduleResolution": "bundler",`. Omitting this entry in tsconfig fixes it indeed, but I'd rather keep it. Any idea on how to fix this?

**Reply by LeunensMichiel** [reply]
Importing from `three/addons/loaders/STLLoader.js` does not work btw.

**Reply by Methuselah96** [reply]
Does it help any if you include the extension (i.e., `import { STLLoader } from "three/examples/jsm/loaders/STLLoader.js";`)?

**Reply by Methuselah96** [reply]
It very likely has to do with [the addition of the `"exports"` entry](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/64877/files#diff-c909a5287d14a671014360f3325189e7f144da59a404f840a036d4d9208a008bR3-R13) in the `"package.json"` in @types/three@0.150.0. With the "exports" defined by three.js, you can no longer import from the examples without specifying the extension.

**Reply by LeunensMichiel** [reply]
That worked perfectly! I didn't know that, learned something new today. Thanks for helping out!
